demo/api/src/warnings/warning-checker.console.ts

```diff
@@ -33,93 +26,59 @@ export class WarningCheckerConsole {
-            baseQueryBuilder
-                .select(["id", ...rootBlockData.map(({ column }) => column)])
-                .from(tableName)
-                .limit(queryBuilderLimit);
-            let rootBlocks: Array<{ [key: string]: BlockInput }> = [];
-            let offset = 0;
-
-            do {
-                const queryBuilder = baseQueryBuilder.clone();
-                queryBuilder.offset(offset);
-                rootBlocks = (await queryBuilder.getResult()) as Array<{ [key: string]: BlockInput }>;
-
-                for (const { column, block } of rootBlockData) {
-                    for (const rootBlock of rootBlocks) {
-                        const blockData = block.blockDataFactory(block.blockInputFactory(rootBlock[column]));
-
-                        const flatBlocks = new FlatBlocks(blockData, {
-                            name: block.name,
-                            visible: true,
-                            rootPath: "root",
-                        });
-                        for (const node of flatBlocks.depthFirst()) {
-                            const warnings = node.block.warnings();
-
-                            if (warnings.length > 0) {
-                                // TODO: (in the next PRs) auto resolve warnings
-                                for (const warning of warnings) {
-                                    const type = "Block";
-                                    const staticNamespace = "4e099212-0341-4bc8-8f4a-1f31c7a639ae";
-                                    const id = v5(`${tableName}${rootBlock["id"]};${warning.message}`, staticNamespace);
-                                    // TODO: (in the next PRs) add blockInfos/metadata
-
-                                    await this.entityManager.upsert(
-                                        Warning,
-                                        {
-                                            createdAt: new Date(),
-                                            updatedAt: new Date(),
-                                            id,
-                                            type,
-                                            message: warning.message,
-                                            severity: WarningSeverity[warning.severity],
-                                        },
-                                        { onConflictExcludeFields: ["createdAt"] },
-                                    );
-                                }
-                            }
-                        }
-                    }
-                }
-
-                offset += queryBuilderLimit;
-            } while (rootBlocks.length > 0);
+        const indexSelects: string[] = [];
+        for (const rootBlockEntity of this.discoverService.discoverRootBlocks()) {
+            const { metadata, column, graphqlObjectType } = rootBlockEntity;
+            const primary = metadata.primaryKeys[0];
+
+            const select = `SELECT
```

Comment:
For dependencies we work with views, is this something we should consider here as well?